__init__.py

```diff
@@ -143,8 +143,10 @@ def initialize(self):
         SkillApi.connect_bus(self.bus)
         self._load_skill_apis()
 
-        self.schedule_repeating_event(self.update_weather, callback_time, 900)
-        self.schedule_repeating_event(self.update_examples, callback_time, 900)
+        self.schedule_repeating_event(self.update_weather, callback_time,
+                                      self.weather_update_seconds)
+        self.schedule_repeating_event(self.update_examples, callback_time,
```

Comment:
this should not use repeating event anymore, otherwise skill setting changes need a restart. we should use a regular event and schedule the next one after executing (re-reading from settings at this time to handle changes)